Tella/Data/Networking/Repositories/GDriveRepository.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  GDriveRepository.swift
+//  Tella
+//
+//  Created by gus valbuena on 5/28/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import GoogleSignIn
+import GoogleAPIClientForREST
+
+protocol GDriveRepositoryProtocol {
+    func handleSignIn(completion: @escaping (Result<Void, Error>) -> Void)
+    func restorePreviousSignIn(completion: (() -> Void)?)
+    func handleUrl(url: URL)
+    func getSharedDrives(completion: @escaping (Result<[SharedDrive], Error>) -> Void)
+}
+
+class GDriveRepository: GDriveRepositoryProtocol  {
+    static let shared = GDriveRepository()
+    
+    private var googleUser: GIDGoogleUser?
+    
+    private var rootViewController: UIViewController? {
+        return UIApplication.shared.windows.first?.rootViewController
+    }
+    
+    func handleSignIn(completion: @escaping (Result<Void, Error>) -> Void) {
```

Comment:
I suggest to use `async await` instead of `completion`
`async await` simplifies the code and makes it more readable by allowing you to write asynchronous code in a synchronous style